### DISCUSSION TITLE
Why aren't these TypeVars assignable?

### DISCUSSION THREAD

**Discussion by d-tw** [microsoft/pyright#1626] [Q&A]
Hi all,

I'm not sure if this is a bug, or something I just don't yet understand about how the type vars work

I don't understand the following error. Can anyone help me understand?

Thanks!

```
Found 1 source file
(...)/e_type.py
  (...)/e_type.py:13:38 - error: Argument of type "Type[E@C]" cannot be assigned to parameter "enum" of type "Type[E@coercer_method]" in function "coercer_method"
    Type "Type[E@C]" cannot be assigned to type "Type[E@coercer_method]"
      Type "E@C" is incompatible with bound type "Type[F]" for TypeVar "E@coercer_method"
        Type "E@C" cannot be assigned to type "Type[F]" (reportGeneralTypeIssues)
1 error, 0 warnings, 0 infos 
```

<img width="578" alt="image" src="https://user-images.githubusercontent.com/3317531/110984525-891c3a80-836b-11eb-8276-7f6f5626987c.png">

```python
from typing import TypeVar, Type, Generic, Union

class F: ...

E = TypeVar('E', bound=F)

def coercer_method(value: Union[E, str], enum: Type[E]) -> E: ...

class C(Generic[E]):
    e_type: Type[E]

    def coerce(self, value: Union[E, str]) -> E:
        return coercer_method(value, self.e_type)
```

**Comment by erictraut** [comment]
Looks like you've found a bug. Could you file it as an issue? I'll take a look at it shortly.

**Reply by erictraut** [reply]
I addressed the problem, and the fix will be in the next release. No need to file a separate bug. And thanks for bringing this to our attention!

**Reply by d-tw** [reply]
Great, thanks!
